crates/sui-sdk/examples/transaction_builder_api.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+mod utils;
+use shared_crypto::intent::Intent;
+use sui_keys::keystore::{AccountKeystore, InMemKeystore, Keystore};
+use sui_sdk::{
+    rpc_types::SuiTransactionBlockResponseOptions,
+    types::{quorum_driver_types::ExecuteTransactionRequestType, transaction::Transaction},
+    SuiClientBuilder,
+};
+use utils::request_tokens_from_faucet;
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let sui = SuiClientBuilder::default().build_localnet().await?;
+
+    // Generate two Sui addresses and corresponding keys in memory
+    let keystore = Keystore::InMem(InMemKeystore::new_insecure_for_tests(2));
+    let addresses = keystore.addresses();
+    let sender = addresses
+        .get(0)
+        .expect("Keystore did not manage to generate two keys in memory. Aborting");
+    let recipient = addresses
+        .get(1)
+        .expect("Keystore did not manage to generate two keys in memory. Aborting");
+
+    // Search for the coins in the sender's address
+    let coins = sui
+        .coin_read_api()
+        .get_coins(*sender, None, None, None)
+        .await?;
+
+    println!("Address {sender} has {} coins", coins.data.len());
+    println!();
+    println!();
+    if coins.next_cursor.is_none() {
+        // Add coins to the newly created sender_address
+        request_tokens_from_faucet(*sender).await?;
+    }
+    let Some(coin) = sui
+        .coin_read_api()
+        .get_coins(*sender, None, None, None)
+        .await?.next_cursor else {panic!("Faucet did not work correctly and the provided Sui address has no coins")};
```

Comment:
nit: formatting (`cargo fmt` is not aware of `let ... else` syntax so you need to help it out with that).